spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1516,10 +1523,16 @@ private void doInvokeBatchOnMessage(final ConsumerRecords<K, V> records,
 			}
 		}
 
-		private void invokeBatchErrorHandler(final ConsumerRecords<K, V> records, RuntimeException e) {
+		private void invokeBatchErrorHandler(final ConsumerRecords<K, V> records,
+				@Nullable List<ConsumerRecord<K, V>> list, RuntimeException e) {
+
 			if (this.batchErrorHandler instanceof ContainerAwareBatchErrorHandler) {
 				this.batchErrorHandler.handle(decorateException(e), records, this.consumer,
-						KafkaMessageListenerContainer.this.thisOrParentContainer);
+						KafkaMessageListenerContainer.this.thisOrParentContainer,
+						() -> {
+							invokeBatchOnMessage(records, list);
+							return true;
```

Comment:
Why do we need this contract as a `Consumer<Boolean>` since we always return `true`?
Won't plain `Runnable` callback be enough?

Reply:
Correct; will change.